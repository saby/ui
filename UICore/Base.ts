/**
 * @kaizen_zone 7d860f70-e142-4269-a5a7-7e940b8be4da
 */
/**
 * Библиотека базового контрола
 * @library UICore/Base
 * @public
 * @includes Control UICore/_base/Control
 */

import { logger } from 'Application/Env';
import { injectHook } from './_base/DevtoolsHook';

injectHook();

// TODO: Реализовать или не использовать.
export function startApplication(cfg?: Record<string, unknown>): void {
    logger.error('В сборке на Реакте нет метода startApplication');
}

/**
 * Возвращаем ноду, от которой начинаем строить.
 * UIReact строит в переданном контейнере
 * @param node Element
 */
export function selectRenderDomNode(node: HTMLElement): HTMLElement {
    return node;
}

export { default as Control, IControlChildren, IControlConstructor } from './_base/Control';
export { UpdatePreventer } from './_base/UpdatePreventer';
export { default as isWasabyControl } from './_base/Control/IWasabyControl';
export { ErrorViewer } from './_base/ErrorViewer';
export { IErrorViewer, TWasabyOverReactProps } from './_base/interfaces';
export { RenderErrorDecorator } from './_base/Control/RenderErrorDecorator';
export { default as RouteCompatible } from './_base/RouteCompatible';
export { default as purifyInstance } from './_base/Control/Purifier/purifyInstance';
export { ASYNC_LOADING_TEST_INDICATOR as AsyncIndicatorName } from './_base/Control/LoaderIndicator';
export { getGeneratorConfig } from 'UICommon/Base';
export { setScrollOnBody } from './_base/IntersectionObserver';
