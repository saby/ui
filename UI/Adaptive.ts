/**
 * Библиотека для настройки адаптивности сайта к различным аспектам адаптивности
 * @see https://n.sbis.ru/wasaby/knowledge#toc_2eb25e0a-c867-45a8-a13d-b8c1ff12cdbc
 * @library UICore/Adaptive
 * @public публичная библиотека
 */

export {
    DEFAULT_BREAKPOINTS,
    AdaptiveContainer,
    AdaptiveInitializerConfig,
    AdaptiveModeType,
    useAdaptiveMode,
    withAdaptiveMode,
    getAdaptiveModeForLoaders,
    unsafe_getRootAdaptiveMode,
    ScrollOnBodyStore,
    BodyScroll,
    moveBodyScroll,
    unMoveBodyScroll,
    ScrollOnBodyContext,
    ScrollOnBodyContextProvider,
} from 'UICore/Adaptive';
