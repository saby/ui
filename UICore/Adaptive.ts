/**
 * @kaizen_zone 4fd9ac53-4889-442d-adee-a7756f91e01b
 */
/**
 * Библиотека для настройки адаптивности
 * @library UICore/Adaptive
 * @public публичная библиотека
 */

export { AdaptiveModeType } from './_adaptive/AdaptiveModeClass';
export { AdaptiveInitializer as AdaptiveInitializerInternal } from './_adaptive/AdaptiveInitializer';
export { DEFAULT_BREAKPOINTS } from './_adaptive/Aspects';
export { AdaptiveContainer } from './_adaptive/AdaptiveContainer';
export {
    getAdaptiveModeForLoaders,
    unsafe_getRootAdaptiveMode,
} from './_adaptive/AdaptiveInitializer';

export { AdaptiveModeClass as _AdaptiveModeClass } from './_adaptive/AdaptiveModeClass';
export { WindowSizeTracker as _WindowSizeTracker } from './_adaptive/SizeTracker';
export { useAdaptiveMode, withAdaptiveMode } from './_adaptive/withAdaptiveMode';
export { applyBodyScroll, moveBodyScroll, unMoveBodyScroll } from './_adaptive/ScrollOnBody';
export { ScrollOnBodyContext } from './_adaptive/ScrollOnBodyContext';
export { ScrollOnBodyContextProvider } from './_adaptive/ScrollOnBodyContextProvider';
export { default as ScrollOnBodyStore } from './_adaptive/ScrollOnBodyStore';
export { Storage, IAspects } from './_adaptive/Aspects';
export { default as BodyScroll } from './_adaptive/BodyScrollHOC';
export { AdaptiveInitializerConfig } from './_adaptive/AdaptiveInitializerConfig';
// только для тестов
export { AdaptiveInitializerConfigContext as _AdaptiveInitializerConfigContext } from './_adaptive/AdaptiveInitializerConfigContext';
