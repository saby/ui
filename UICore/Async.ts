/**
 * @kaizen_zone 4cbeec99-a5a5-48bf-8531-411cf558b0c0
 */
/**
 * Библиотека абстрактной реализации контейнера для асинхронной загрузки контролов.
 * @library UICore/Async
 * @public
 * @includes Async UICore/_async/Async
 */

export { default as Async, IAsyncOptions, TAsyncStateReceived } from 'UICore/_async/Async';
