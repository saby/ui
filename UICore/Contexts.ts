/**
 * @kaizen_zone aed53143-c89f-413e-a8e4-de8b6ad43abe
 */
/**
 * @library
 * @incudes WasabyContextManager UICore/_contexts/WasabyContextManager
 * @incudes withTheme UICore/_contexts/withTheme
 * @incudes withReadonly UICore/_contexts/withReadonly
 * @incudes WasabyContext UICore/_contexts/WasabyContext
 * @public
 */
export { WasabyContextManager, wasabyContextPropNames } from './_contexts/WasabyContextManager';
export { useTheme, withTheme } from './_contexts/withTheme';
export { useReadonly, withReadonly } from './_contexts/withReadonly';
export { getWasabyContext, IWasabyContextValue, TWasabyContext } from './_contexts/WasabyContext';
