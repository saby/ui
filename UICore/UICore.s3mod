<?xml version="1.0" encoding="UTF-8" ?>
<ui_module id="548e4ee9-10cd-4227-9e3c-0811958f979c" name="UICore" kaizen_zone="b1b57bba-bc4d-4efe-9e96-403a7c5fdafc" is_react="1" package="Wasaby" responsible="Шипин А.А." responsible_uuid="8619c702-3e02-4c65-9cab-002183de664a" required="1" version="1.00">

  <description>Библиотека для работы ядра через React</description>

  <depends>
    <module id="0df0fce3-53ba-47f8-987d-edf1fd078295" name="WS.Core"/>
    <module id="636177e5-4d75-44fa-9173-977dc90ff111" name="Env"/>
    <module id="323fd54f-1fab-42a6-8aea-6726d6392d53" name="Application"/>
    <module id="686f183d-0bcc-4732-b4e4-6016c2a67a0f" name="Types"/>
    <module id="a5e32491-7b02-45b1-b65d-53e9723ec0cd" name="React"/>
    <module id="0f15ac6b-86b2-4265-9da1-05eac93ebdf6" name="UICommon" />
    <module id="2274d173-fe30-4fcf-81ca-767915919d69" name="EnvTouch"/>
    <module id="6ae4f1d6-9ecf-4805-a98c-65e47906eedd" name="Tailwind"/>
    <module id="4faefc42-6d2d-4c62-8a73-8de7ae255183" name="DefaultThemeController" />
  </depends>

  <load_after/>

  <parameters/>

  <resources/>

  <preload/>

  <typescript>
    <arg key="typecheck" value="0"/>
  </typescript>
</ui_module>
