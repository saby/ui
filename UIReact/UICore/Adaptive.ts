/**
 * Библиотека для настройки адаптивности
 * @library UICore/Adaptive
 * @public публичная библиотека
 */

export { AdaptiveModeType } from './_adaptive/AdaptiveModeClass';
export { AdaptiveInitializer } from './_adaptive/AdaptiveInitializer';
export { DEFAULT_BREAKPOINTS } from './_adaptive/AdaptiveModeClass';
export { AdaptiveContainer } from './_adaptive/AdaptiveContainer';
export {
    getAdaptiveModeForLoaders,
    unsafe_getRootAdaptiveMode,
} from './_adaptive/AdaptiveInitializer';

export { AdaptiveModeClass as _AdaptiveModeClass } from './_adaptive/AdaptiveModeClass';
export { WindowSizeTracker as _WindowSizeTracker } from './_adaptive/SizeTracker';
export { useAdaptiveMode, withAdaptiveMode } from './_adaptive/withAdaptiveMode';
export { applyBodyScroll } from './_adaptive/ScrollOnBody';
