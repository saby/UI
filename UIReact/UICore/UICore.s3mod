<?xml version="1.0" encoding="UTF-8" ?>
<ui_module id="548e4ee9-10cd-4227-9e3c-0811958f979c" name="UICore" package="Платформа" responsible="Шипин А.А." required="1" version="1.00">

  <description>Библиотека для работы ядра через React. Имеет общее API с UIInferno/UICore</description>

  <depends>
    <module id="0f15ac6b-86b2-4265-9da1-05eac93ebdf6" name="UICommon" />
  </depends>

  <load_after/>

  <parameters/>

  <resources/>

  <preload/>

</ui_module>
