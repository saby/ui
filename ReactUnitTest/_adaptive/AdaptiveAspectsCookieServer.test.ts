import { logger, cookie } from 'Application/Env';
import { unsafe_getRootAdaptiveMode, Storage } from 'UICore/Adaptive';

describe('adaptiveAspects cookie server', () => {
    let currentAdaptiveAspectsCookie: string;
    let errorSpy: jest.SpyInstance;
    beforeEach(() => {
        errorSpy = jest.spyOn(logger, 'error').mockImplementation(() => {});

        const originalCookieGet = cookie.get;
        jest.spyOn(cookie, 'get').mockImplementation((key) => {
            if (key === 's3ac') {
                return currentAdaptiveAspectsCookie;
            }
            return originalCookieGet(key);
        });

        const originalCookieSet = cookie.set;
        jest.spyOn(cookie, 'set').mockImplementation((key, value, options) => {
            if (key === 's3ac') {
                currentAdaptiveAspectsCookie = value;
                return true;
            }
            return originalCookieSet(key, value, options);
        });
    });
    afterEach(() => {
        currentAdaptiveAspectsCookie = null;
        jest.restoreAllMocks();

        Storage.getInstance()._clear();
    });

    test('valid', () => {
        // '{"windowInnerWidth":31415,"windowInnerHeight":31415,"isPhone":true,"isTouch":true}'
        currentAdaptiveAspectsCookie = '31415!31415!!!!!!!!!10';
        const rootAdaptiveMode = unsafe_getRootAdaptiveMode();

        expect(JSON.stringify(rootAdaptiveMode, undefined, ' ')).toMatchSnapshot();
        // '{"windowInnerWidth":31415,"windowInnerHeight":31415,"isPhone":false,"isTouch":true,"isTablet":false}'
        expect(currentAdaptiveAspectsCookie).toBe('31415!31415!!!!!!!!!10');
        expect(errorSpy).toBeCalledTimes(0);
    });
    test('invalid', () => {
        currentAdaptiveAspectsCookie = '../../../../../../../../../../../../../../etc/passwd';
        const rootAdaptiveMode = unsafe_getRootAdaptiveMode();

        expect(JSON.stringify(rootAdaptiveMode, undefined, ' ')).toMatchSnapshot();
        expect(currentAdaptiveAspectsCookie).toBe('!!!!!!!!!!0');
        expect(errorSpy).toBeCalledTimes(0);
    });
    test('set', () => {
        Storage.getInstance().set({
            windowInnerWidth: 31415,
            windowInnerHeight: 31415,
            isPhone: true,
            isTouch: true,
        });

        expect(Storage.getInstance().get()).toMatchSnapshot();
    });
    test('set with dots', () => {
        Storage.getInstance().set({
            windowInnerWidth: 123.123,
            windowInnerHeight: 123.123,
            windowOuterWidth: 123.123,
            windowOuterHeight: 123.123,
            viewportWidth: 123.123,
            viewportHeight: 123.123,
            screenWidth: 123.123,
            screenHeight: 123.123,
            containerClientWidth: 123.123,
            containerClientHeight: 123.123,
        });

        expect(Storage.getInstance().get()).toMatchSnapshot();
    });
});
