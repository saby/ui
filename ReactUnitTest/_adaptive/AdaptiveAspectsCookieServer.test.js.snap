// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adaptiveAspects cookie server invalid 1`] = `
"{
 \\"$serialized$\\": \\"inst\\",
 \\"module\\": \\"UICore/Adaptive:_AdaptiveModeClass\\",
 \\"id\\": 2,
 \\"state\\": {
  \\"containerClientWidth\\": 768,
  \\"containerClientHeight\\": 1024
 }
}"
`;

exports[`adaptiveAspects cookie server set 1`] = `Object {}`;

exports[`adaptiveAspects cookie server set with dots 1`] = `Object {}`;

exports[`adaptiveAspects cookie server valid 1`] = `
"{
 \\"$serialized$\\": \\"inst\\",
 \\"module\\": \\"UICore/Adaptive:_AdaptiveModeClass\\",
 \\"id\\": 1,
 \\"state\\": {
  \\"containerClientWidth\\": 768,
  \\"containerClientHeight\\": 1024
 }
}"
`;
